src/api.js

```diff
@@ -2087,6 +2087,79 @@ api.declare({
   return res.reply(result);
 });
 
+/** Get a worker-type from a provisioner */
+api.declare({
+  method:     'get',
+  route:      '/provisioners/:provisionerId/worker-types/:workerType',
+  name:       'getWorkerType',
+  stability:  API.stability.experimental,
+  output:     'workertype-response.json#',
+  title:      'Get a worker-type',
+  description: [
+    'Get a worker-type from a provisioner.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {provisionerId, workerType} = req.params;
+
+  const wType = await this.WorkerType.scan({
+    provisionerId,
+    workerType,
+    expires: Entity.op.greaterThan(new Date()),
+  });
+
+  const result = wType.entries.length ? wType.entries[0].json() : {};
+
+  return wType.entries.length ?
+    res.reply(result) :
+    res.reportError('ResourceNotFound',
+      'Worker-type {{workerType}} with  Provisioner {{provisionerId}} not found. Are you sure it was created?', {
+        workerType,
+        provisionerId,
+      },
+    );
+});
+
+/** Update a worker-type */
+api.declare({
+  method:     'put',
+  route:      '/provisioners/:provisionerId/worker-types/:workerType',
+  name:       'updateWorkerType',
+  stability:  API.stability.experimental,
+  scopes:     [
+    [
+      'queue:declare-worker-type:<provisionerId>/<workerType>#<property>',
+    ],
+  ],
+  output:     'workertype-response.json#',
+  input:      'update-workertype-request.json#',
+  title:      'Update a worker-type',
+  description: [
+    'Update a worker-type from a provisioner.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {provisionerId, workerType} = req.params;
+  const {stability, description, expires} = req.body;
+
+  const wType = await this.WorkerType.load({provisionerId, workerType}, true);
+
+  if (!wType) {
+    res.reportError('ResourceNotFound',
+      'Worker-type {{workerType}} with  Provisioner {{provisionerId}} not found. Are you sure it was created?', {
+        workerType,
+        provisionerId,
+      },
+    );
+  }
+
+  const result = await wType.modify((entity) => {
```

Comment:
This will need some scopes checked first!  I liked Jonas's suggestion:
```
scopes:    'queue:declare-worker-type:<provisionerId>/<workerType>#<property>'
```
with the property checked for each property included in the request.  Then we can control who can update which property.

Reply:
I declared `scopes` in the options: https://github.com/taskcluster/taskcluster-queue/pull/190/commits/9a59d3bc5dccc665562b989d86430e49c0a5ab80#diff-e1d62d8c4965edf0e532dfa19da0891fR2128 wouldn't that do it?

Also, just out of curiosity, could you give me an example of `<property>`

Reply:
Oh, I did miss that.  But we need more, since the framework isn't going to know what to do with `<property>`.  You'll need to add the `deferAuth` option and call req.satisfies a few times, once for each property.  `<property>` in this case would be `stability`, `description`, or `expires`.

Reply:
Makes sense, thanks for the clarifications.